packages/core/src/messagingProtocols/mls/E2eIdentityService/Connection/AcmeServer/AcmeService.ts

```diff
@@ -0,0 +1,199 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {AcmeChallenge, AcmeDirectory} from '@wireapp/core-crypto/platforms/web/corecrypto';
+import axios, {AxiosInstance} from 'axios';
+import logdown from 'logdown';
+
+import {
+  GetDirectoryReturnValue,
+  GetInitialNonceReturnValue,
+  PostJoseRequestParams,
+  PostJoseRequestReturnValue,
+} from './AcmeService.types';
+import {
+  AuthorizationResponseData,
+  AuthorizationResponseSchema,
+  DirectoryResponseSchema,
+  NewAccountResponseData,
+  NewAccountResponseSchema,
+  NewOrderResponseData,
+  NewOrderResponseSchema,
+  ResponseHeaderLocation,
+  ResponseHeaderLocationSchema,
+  ResponseHeaderNonce,
+  ResponseHeaderNonceSchema,
+  CheckStatusOfOrderResponseData,
+  CheckStatusOfOrderResponseSchema,
+  DpopChallengeResponseData,
+  DpopChallengeResponseSchema,
+  FinalizeOrderResponseData,
+  FinalizeOrderResponseSchema,
+  OidcChallengeResponseData,
+  OidcChallengeResponseSchema,
+  GetCertificateResponseData,
+  GetCertificateResponseSchema,
+} from './schema';
+
+export class AcmeService {
+  private logger = logdown('@wireapp/core/AcmeService');
+  private readonly axiosInstance: AxiosInstance = axios.create();
+  private readonly url = {
+    DIRECTORY: '/directory',
+  };
+
+  constructor(private discoveryUrl: string) {}
+
+  // ############ Internal Functions ############
+
+  private extractNonce(headers: any): ResponseHeaderNonce['replay-nonce'] {
+    return ResponseHeaderNonceSchema.parse(headers)['replay-nonce'];
+  }
+
+  private extractLocation(headers: any): ResponseHeaderLocation['location'] {
+    return ResponseHeaderLocationSchema.parse(headers).location;
+  }
+
+  private async postJoseRequest<T>({
+    payload,
+    schema,
+    url,
+    errorMessage,
+    shouldGetLocation = false,
+  }: PostJoseRequestParams<T>): PostJoseRequestReturnValue<T> {
+    try {
+      const {data, headers} = await this.axiosInstance.post(url, payload, {
+        headers: {
+          'Content-Type': 'application/jose+json',
```

Comment:
Let's add `'application/jose+json'` to `ContentType.ts` file in  `api-client`. Also I think requests should be part of `api-client` not `core`.

Reply:
I don't see it this way.

The api-Client package encapsulates all requests to the wire server.

We don't access the wire server here; we access a particular and different environment.
Thematically all of these calls belong to E2EI.

All requests that utilise the wire server are located in api-client.

Reply:
OK, if we assume that `api-client` is making requests only to our own api, then it's fine 👍